recipes/liquid-dsp/all/conanfile.py

```diff
@@ -0,0 +1,172 @@
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+
+    def _gen_link_library(self):
+        if self.settings.compiler != "Visual Studio" or (not self.options.shared):
+            return
+        self.run("cmd /c generate_link_library.bat")
+        with tools.chdir(self._source_subfolder):
+            self.run(
+                "{} /def:libliquid.def /out:libliquid.lib /machine:{}".format(
+                    os.getenv("AR"), "X86" if self.settings.arch == "x86" else "X64"
+                ),
+                win_bash=tools.os_info.is_windows,
+            )
+
+    def _rename_libraries(self):
+        with tools.chdir(self._source_subfolder):
+            if self.settings.os == "Windows" and self.options.shared:
+                tools.rename("libliquid.so", "libliquid.dll")
+            elif self.settings.os == "Windows" and not self.options.shared:
+                tools.rename("libliquid.a", "libliquid.lib")
+            elif self.settings.os == "Macos" and not self.options.shared:
+                tools.rename("libliquid.ar", "libliquid.a")
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            env = {
+                "CC": "gcc",
+                "CXX": "g++",
+                "LD": "ld",
+                "AR": "ar",
+            }
+            with tools.environment_append(env):
+                yield
+        else:
+            yield
+
+    @contextmanager
+    def _msvc_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self.settings):
+                env = {
+                    "CC": "cl -nologo",
+                    "CXX": "cl -nologo",
+                    "AR": "lib",
+                    "LD": "link",
+                }
+                with tools.environment_append(env):
+                    yield
+        else:
+            yield
+
+    def build(self):
+        self._patch_sources()
+        ncpus = tools.cpu_count()
+        configure_args = []
+        cflags = ["-static-libgcc"]
+        if self.settings.build_type == "Debug":
+            configure_args.append("--enable-debug-messages")
+            cflags.extend(["-g", "-O0"])
+        else:
+            cflags.extend(["-s", "-O2", "-DNDEBUG"])
+        if self.options.simdoverride:
+            configure_args.append("--enable-simdoverride")
+        if self.settings.compiler == "Visual Studio":
+            configure_args.append("CFLAGS='{}'".format(" ".join(cflags)))
+        configure_args_str = " ".join(configure_args)
+        with self._build_context():
+            with tools.chdir(self._source_subfolder):
+                self.run("./bootstrap.sh", win_bash=tools.os_info.is_windows)
+                self.run(
+                    "./configure {}".format(configure_args_str),
+                    win_bash=tools.os_info.is_windows,
+                )
```

Comment:
Same boat this is a pain to build... but I don't want to block it since it's useable

Typically I see the experts do https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/gnulib/all/test_package/conanfile.py#L44

Then they can use the autotools helper for the configure and build